### PR TITLE
docs: add documentation using jsdoc

### LINKED ISSUE #39: Add script to convert comments to docs using jsdoc
Create an npm script `npm docs` that will export all the comments and make it available as html page at `/docs/reference` endpoint. We are already writing jsdoc supported comments, use [jsdoc npm package](https://jsdoc.app/) now to convert those comments to html

I think, following should fix it
1. Install jsdoc as dev dependency `npm install --save-dev jsdoc`
2. Create docs-config.json with source including all js files in the project and destination to be `public/docs/references`  (read about jsdoc configs [here](https://jsdoc.app/about-configuring-jsdoc.html#default-configuration-options))
3. Add script in package.json `scripts : { docs : "jsdoc -c ./docs-config.json" **}`**
4. Test the route `/docs/references`
5. Look for further improvements in docs navigation and then commit the docs folder

Resources 
- [jsdoc command line references](https://jsdoc.app/about-commandline.html)

### PR DESCRIPTION
**Pull request by Dun-sin**
## Changes Proposed
- add config file for jsdoc
- create documentation using jsdoc

### CODE REVIEW

File: public/docs/scripts/prettify/lang-css.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+PR.registerLangHandler(PR.createSimpleLexer([["pln",/^[\t\n\f\r ]+/,null," \t\r\n"]],[["str",/^"(?:[^\n\f\r"\\]|\\(?:\r\n?|\n|\f)|\\[\S\s])*"/,null],["str",/^'(?:[^\n\f\r'\\]|\\(?:\r\n?|\n|\f)|\\[\S\s])*'/,null],["lang-css-str",/^url\(([^"')]*)\)/i],["kwd",/^(?:url|rgb|!important|@import|@page|@media|@charset|inherit)(?=[^\w-]|$)/i,null],["lang-css-kw",/^(-?(?:[_a-z]|\\[\da-f]+ ?)(?:[\w-]|\\\\[\da-f]+ ?)*)\s*:/i],["com",/^\/\*[^*]*\*+(?:[^*/][^*]*\*+)*\//],["com",
```
[the remark this answers is not in the document]
Comment (makeitcount) [reply public/docs/scripts/prettify/lang-css.js:1]: @Dun-sin is it possible to fix this without breaking the functionality?
Reference to this security alert - https://codeql.github.com/codeql-query-help/javascript/js-redos

**Comment by makeitcount** [comment]
@Dun-sin requested few minor changes. Please review. Other than that, it looks good.

File: public/docs/scripts/prettify/lang-css.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Dun-sin) [reply public/docs/scripts/prettify/lang-css.js:1]: This doesn't have to do with code, it's just concerned with the docs and doesn't touch the actual code. This code was auto-generated by Jsdoc

**Comment by makeitcount** [comment]
@Dun-sin your repository seems to be deleted. Wondering if it was intentional? Is there anything I can help with in order to take this one ahead and merge?

**Comment by Dun-sin** [comment]
Yep it was intentional, totally forgot about this😅

I really don't know what's with the Github security errors tho

**Pull request closed without merging**